### PR TITLE
[DL-351] Small change on FlowIo calculator to store allocations within order's flow data.

### PR DESCRIPTION
**Pull request by sebastiandl**
### What problem is the code solving?
When creating the CI for Flow orders we have to fill in taxes and duties information.

### How does this change address the problem?
In order to avoid having to call again Flow to retrieve this information when creating the CI, we are storing it within the order's data when retrieving it for adjustment's calculation.

### Why is this the best solution?
Storing this information within the order's data will help us avoid multiple calls to Flow to retrieve this information.

### Share the knowledge
Nothing to add

### CODE REVIEW

File: app/models/spree/calculator/flow_io.rb [hunk 1]
```diff
@@ -20,6 +20,13 @@ def compute_shipment_or_line_item(item)
       alias compute_shipment compute_shipment_or_line_item
       alias compute_line_item compute_shipment_or_line_item
 
+      def get_tax_rate(taxable)
```
Comment (sebastiandl) [inline app/models/spree/calculator/flow_io.rb:23]: Currently being used by AuditTax model when auditing the tax for Flow.

File: app/models/spree/flow_io_variant_decorator.rb [hunk 2]
```diff
@@ -50,15 +50,16 @@ def remove_experience_from_product(exp_key, product)
     end
 
     # upload product variant to Flow's Product Catalog
-    def sync_product_to_flow
+    def sync_product_to_flow(force_sync = false)
```
Comment (earthtodani) [inline app/models/spree/flow_io_variant_decorator.rb:53]: What does force sync represent?

**Review by earthtodani — APPROVED** [review]

File: app/models/spree/flow_io_variant_decorator.rb (hunk 2 above)
Comment (sebastiandl) [reply app/models/spree/flow_io_variant_decorator.rb:53]: This **force_sync** param would be use to help us force the synchrornization of some products that does not match the minimum requirements by Flow (price more than 0 or having country of origin).

Right now, for example we have the **Ring Sizer**, which is not being charged at all; However, we still want to offer it on Flow countries. In order to do so, we need to force that product's synchronization into Flow and then set an Euro price for it using price books inside Flow.

I also realized that maybe I can rearrange this better, by separating this method into several ones and have a new one named after (force_sync_product_to_flow) that would skip validations.

Thanks for asking

Comment (earthtodani) [reply app/models/spree/flow_io_variant_decorator.rb:53]: > I also realized that maybe I can rearrange this better, by separating this method into several ones and have a new one named after (force_sync_product_to_flow) that would skip validations.

Yep, I agree! It is a code smell to pass [boolean as a parameter](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3) as it is a sign of a method doing too much. 
There is also a [check in reek](https://github.com/troessner/reek/blob/master/docs/Boolean-Parameter.md) but it is disabled now, maybe we should enable it?

**Pull request merged**